File: 3768-check-if-digits-are-equal-in-string-after-operations-i/check-if-digits-are-equal-in-string-after-operations-i.cpp
class Solution {
public:
    bool hasSameDigits(string s) {
        while(s.length()!=2){
            string temp="";
            int i=0;
            int j=1;
            while(j<s.length()){
                temp.push_back((s[i]-'0' + s[j]-'0')%10 +'0');
                i++;
                j++;
            }
            s=temp;
        }
        return s[0]==s[1];
    }
};

File: 3768-check-if-digits-are-equal-in-string-after-operations-i/check-if-digits-are-equal-in-string-after-operations-i.java
class Solution {
    public boolean hasSameDigits(String s) {
        int n = s.length();
        char[] arr = s.toCharArray();

        while (n > 2) {
            for (int i = 0; i < n - 1; i++) {
                int num = (arr[i] - '0' + arr[i + 1] - '0') % 10;
                arr[i] = (char)(num + '0');
            }
            n--;
        }

        return arr[0] == arr[1];
    }
}
